library/DataIdentity.cpp

```diff
@@ -49,7 +49,9 @@ namespace df {
     STL_OPAQUE_IDENTITY_TRAITS(mutex);
     STL_OPAQUE_IDENTITY_TRAITS(future<void>);
     STL_OPAQUE_IDENTITY_TRAITS(function<void()>);
-    STL_OPAQUE_IDENTITY_TRAITS(optional<std::function<void()> >);
+    STL_OPAQUE_IDENTITY_TRAITS(optional<void_function>);
+    OPAQUE_IDENTITY_TRAITS_NAME(std::variant<std::string COMMA void_function>,
```

Comment:
there's a betetr way to do this using __VA_ARGS__

Reply:
Thanks. I was pretty unhappy with how this looked

Reply:
might want to cherry-pick bef0cdeb689551dd70afa782c726c5a3008af327, which is on my 5012-testing-1 branch

Reply:
see #4092